File: LeetCodeProblems/14. Longest Common Prefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] words) {
        // Edge case: if the array is empty, return an empty string
        if (words.Length == 0) return "";

        // Start by assuming the first word is the common prefix
        string commonPrefix = words[0];

        // Loop through each word in the array
        foreach (string word in words) {
            // While the current word does not start with the current prefix,
            // shorten the prefix by removing the last character
            while (!word.StartsWith(commonPrefix)) {
                // Remove the last character from the prefix
                commonPrefix = commonPrefix.Substring(0, commonPrefix.Length - 1);

                // If the prefix becomes empty, there is no common prefix
                if (commonPrefix == "") return "";
            }
        }

        // After checking all words, return the longest common prefix found
        return commonPrefix;
    }
}

File: LeetCodeProblems/JavaScript/14. Longest Common Prefix.js
var longestCommonPrefix = function (strs) {
  if (strs.length === 0) return "";
  if (strs.length === 1) return strs[0];

  // Sort strings - common prefix will be between first and last
  strs.sort();
  const first = strs[0];
  const last = strs[strs.length - 1];

  let i = 0;
  // Find common prefix between first and last sorted strings
  while (i < first.length && i < last.length && first[i] === last[i]) {
    i++;
  }

  return first.substring(0, i);
};
